src/components/Dropdowns/StyledSelect.tsx

```diff
@@ -3,14 +3,42 @@ import React, { ReactElement, useMemo } from "react";
 import Select, { components, DropdownIndicatorProps, StylesConfig } from "react-select";
 import { StateManagerProps } from "react-select/dist/declarations/src/useStateManager";
 import styled, { css } from "styled-components";
+import { Color } from "three";
 
 import { DropdownSVG } from "../../assets";
 
 import { AppTheme, AppThemeContext } from "../AppStyle";
 
+type OptionWithColor = unknown & { color?: Color };
+
 type AntStyledSelectProps = StateManagerProps & {
   type?: ButtonProps["type"] | "outlined";
   width?: string;
+  styles?: StylesConfig;
+};
+
+/**
+ * If `color` is defined, adds style config properties  that add a colored
+ * pseudo-element indicator to the left of an option text in the dropdown.
+ */
+const addOptionalColorIndicator = (color: Color | undefined): Partial<StylesConfig["option"]> => {
+  if (color) {
+    return {
+      alignItems: "center",
+      display: "flex",
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      ":before": {
+        backgroundColor: "#" + color.getHexString(),
+        borderRadius: 10,
+        content: '" "',
+        display: "block",
+        marginRight: 8,
+        height: 10,
+        width: 10,
+      },
+    };
+  }
+  return {};
 };
```

Comment:
Derived from the official react-select docs: https://react-select.com/home#custom-styles